bn254/src/fq.rs

```diff
@@ -184,7 +184,7 @@ prime_field_operation!(Fq, MODULUS, GENERATOR, INV, R, R2, R3);
 
 impl From<Fr> for Fq {
     fn from(val: Fr) -> Fq {
-        Self(val.0)
+        Self(to_mont_form(val.montgomery_reduce(), R2, MODULUS, INV))
```

Comment:
I think we don't perform reduce either of them

Reply:
I'll check if it's possible to do without reduction.